codecparsers/mpeg2_parser.cpp

```diff
@@ -356,12 +355,83 @@ namespace MPEG2 {
         DEBUG("time_code_minutes  : %x", m_GOPHeader.time_code_minutes);
         DEBUG("time_code_seconds  : %x", m_GOPHeader.time_code_seconds);
         DEBUG("time_code_pictures : %x", m_GOPHeader.time_code_pictures);
+        DEBUG("closed_gop 	  : %x", m_GOPHeader.closed_gop);
 
         bitReaderDeInit();
 
         return true;
     }
 
+    void Parser::readQuantMatrix(bool& loadMatrix, uint8_t matrix[])
+    {
+
+        bitReaderReadFlag(&loadMatrix);
```

Comment:
So far the nal units will be checked for the smallest size they can have, a broken nal unit will result in artifacts or complete  shutdown of the component depending on what is missing from the stream.  Some of them will be non-fatal and decoding can continue. If you agree a more strict size enforcement can be made in the future with a good categorization on the nature of the missing information.  It will be bad two always shutdown if information is missing on the stream.